### PR TITLE
WIP: [m3collector] implements prometheus remote write api

### PR DESCRIPTION
**Pull request by eahydra**
Hi, we had ran the m3collector with m3aggregator succeed, and we modified the Prometheus code that embedded the m3collector into Prometheus which implements Prometheus Storage Interface.

Now we have separate the code from our internal Prometheus version and merge into the m3collector that implements the Prometheus Remote Write API. And we want to feedback to M3DB

### CODE REVIEW

**Comment by robskillington** [comment]
@eahydra thank you for this!

File: src/collector/api/v1/handler/prometheus/remote/write.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+		}
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *PromWriteHandler) newMetricID(l []*prompb.Label) (id.ID, error) {
```
Comment (robskillington) [inline src/collector/api/v1/handler/prometheus/remote/write.go]: We should probably move this logic to a central place so we don't duplicate this with the existing remote write logic. 

Perhaps we can put it in the `src/cmd/coordinator/downsample` package and export it?

File: src/collector/server/id_pool_types.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+	"github.com/m3db/m3/src/x/serialize"
+	"github.com/m3db/m3x/checked"
+	"github.com/m3db/m3x/ident"
+	"github.com/m3db/m3x/pool"
+
+	"github.com/prometheus/common/model"
+)
+
+var (
+	defaultMetricNameTagName = []byte(model.MetricNameLabel)
```
Comment (robskillington) [inline src/collector/server/id_pool_types.go]: Similar to above in this file, we should probably export logic to a central place so we don't duplicate this with the existing remote write logic. 

Perhaps we can export these types from `src/cmd/coordinator/downsample`?

File: src/collector/api/v1/handler/prometheus/remote/write.go (hunk 1 above)
Comment (eahydra) [reply src/collector/api/v1/handler/prometheus/remote/write.go]: Yeah, I will refer to your comments to make changes.

**Pull request closed without merging**